### PR TITLE
PM-15891: Updated slider

### PR DESCRIPTION
**Pull request by ezimet-livefront**
## 🎟️ Tracking

[PM-15891](https://bitwarden.atlassian.net/browse/PM-15891)
## 📔 Objective

- Created custom ```BitwardenSlider``` that allows for the customization of the thumb view.
- 
## 📸 Screenshots

Screenshot 2024-12-19 at 4 24 20 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-15891]: https://bitwarden.atlassian.net/browse/PM-15891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Platform/Application/Views/FormFields/SliderFieldView.swift [hunk 1]
```diff
@@ -53,31 +53,30 @@ struct SliderFieldView<State>: View {
...
-            .tint(Asset.Colors.tintPrimary.swiftUIColor)
+            ) {
+                Circle()
+                    .fill(Asset.Colors.sliderFilled.swiftUIColor)
+                    .frame(width: 18, height: 18)
+                    .overlay(
+                        Circle()
+                            .stroke(Asset.Colors.sliderThumbBorder.swiftUIColor, lineWidth: 2)
+                    )
+            }
```
[the remark this answers is not in the document]
Comment (ezimet-livefront) [reply BitwardenShared/UI/Platform/Application/Views/FormFields/SliderFieldView.swift:79]: I thought we would not use ```BitwardenSlider``` directly, and use ```SliderFieldView```, but I can have default ```thumbview``` in  ```BitwardenSlider```  and still expose the customization of thumb view.

[the remark this answers is not in the document]
Comment (matt-livefront) [reply BitwardenShared/UI/Platform/Application/Views/FormFields/SliderFieldView.swift:79]: I'm not sure; the `*FieldView` views aren't really used outside of the generator. I think it's better to include it directly in `BitwardenSlider`, I don't really envision us needing to customize that.

[the remark this answers is not in the document]
Comment (ezimet-livefront) [reply BitwardenShared/UI/Platform/Application/Views/FormFields/SliderFieldView.swift]: just moved that to ```BitwardenSlider```.

**Review by matt-livefront — APPROVED** [review]

**Review by fedemkr — COMMENTED** [review]
🎨 Could you add screenshots to the PR description to see the updated control?
🎨 Could you add tests for `BitwardenSlider`?

**Review by fedemkr — APPROVED** [review]

**Review by fedemkr — APPROVED** [review]

**Review by matt-livefront — APPROVED** [review]

**Pull request merged**